apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/LiveAppSDKLogic.ts

```diff
@@ -0,0 +1,391 @@
+import { prepareMessageToSign } from "@ledgerhq/live-common/hw/signMessage/index";
+
+type WebPlatformContext = {
+  manifest: AppManifest;
+  dispatch: Dispatch;
+  accounts: AccountLike[];
+};
+
+function getParentAccount(account: AccountLike, fromAccounts: AccountLike[]): Account | null {
+  if (isAccount(account)) {
+    return null;
+  }
+
+  const parentAccount = fromAccounts.find(a => a.id === account.parentId);
+  if (!parentAccount) {
+    throw new Error("subaccount has no parent account");
+  }
+  if (!isAccount(parentAccount)) {
+    throw new Error("parentAccount is not of type Account");
+  }
+  return parentAccount;
+}
+
+export const receiveOnAccountLogic = (
+  { manifest, dispatch, accounts }: WebPlatformContext,
+  accountId: string,
+) => {
+  tracking.platformReceiveRequested(manifest);
+
+  const account = accounts.find(account => account.id === accountId);
+
+  if (!account) {
+    tracking.platformReceiveFail(manifest);
+    return Promise.reject(new Error("Account required"));
+  }
+
+  const parentAccount = isTokenAccount(account)
+    ? accounts.find(a => a.id === account.parentId)
+    : null;
+
+  // FIXME: handle address rejection (if user reject address, we don't end up in onResult nor in onCancel 🤔)
+  return new Promise((resolve, reject) =>
+    dispatch(
+      openModal("MODAL_EXCHANGE_CRYPTO_DEVICE", {
+        account,
+        parentAccount,
+        onResult: (account: Account, parentAccount: Account) => {
+          tracking.platformReceiveSuccess(manifest);
+          resolve(accountToPlatformAccount(account, parentAccount).address);
+        },
+        onCancel: (error: Error) => {
+          tracking.platformReceiveFail(manifest);
+          reject(error);
+        },
+        verifyAddress: true,
+      }),
+    ),
+  );
+};
+
+export type RequestAccountParams = {
+  currencies?: string[];
+  allowAddAccount?: boolean;
+  includeTokens?: boolean;
+};
+export const requestAccountLogic = async (
+  { manifest }: Omit<WebPlatformContext, "accounts" | "dispatch">,
+  { currencies, includeTokens }: RequestAccountParams,
+) => {
+  tracking.platformRequestAccountRequested(manifest);
+  const { account, parentAccount } = await selectAccountAndCurrency(currencies, includeTokens);
+
+  return serializePlatformAccount(accountToPlatformAccount(account, parentAccount));
+};
+
+export const signTransactionLogic = (
+  { manifest, dispatch, accounts }: WebPlatformContext,
+  accountId: string,
+  transaction: RawPlatformTransaction,
+  params: any,
+) => {
+  tracking.platformSignTransactionRequested(manifest);
+
+  const platformTransaction = deserializePlatformTransaction(transaction);
+
+  const account = accounts.find(account => account.id === accountId);
+
+  if (!account) {
+    tracking.platformSignTransactionFail(manifest);
+    return Promise.reject(new Error("Account required"));
+  }
+
+  const parentAccount = isTokenAccount(account) ? getParentAccount(account, accounts) : null;
```

Comment:
We can probably simplify this IMO, because getParentAccount checks everything IIUC
```suggestion
  const parentAccount = getParentAccount(account, accounts);
```

Reply:
The test are not exactly the same. This is why I asked you about TokenAccount and ChildAccount.
The test in the getParentAccount does not test `isTokenAccount`. It leaves the fact that a `ChildAccount` can be used.

Reply:
Okay I understand better why you asked me that earlier.
But I think the logic is still correct with a `ChildAccount`, I would need confirmation from @IAmMorrow